src/common/services/api/cart.service.js

```diff
@@ -167,8 +167,15 @@ class Cart {
     if (!disableSessionRestart && this.sessionService.getRole() === Roles.public) {
       return this.getTotalQuantity().mergeMap((total) => {
         if (total <= 0) {
-          return this.sessionService.signOut().mergeMap(() => {
-            return this._addItem(uri, data)
+          return this.sessionService.oktaIsUserAuthenticated().mergeMap((isAuthenticated) => {
+            if (!isAuthenticated) {
+              return this._addItem(uri, data)
+            }
+            // SignOut() will redirect user to Okta to clear session,
```

Comment:
@dr-bizz What does this code do? Why are we signing out of Okta when the user is authenticated?